bilge-impl/src/shared.rs

```diff
@@ -173,52 +148,27 @@ pub(crate) fn is_fallback_attribute(attr: &Attribute) -> bool {
     is_attribute(attr, "fallback")
 }
 
-pub(crate) struct EnumVariantValueAssigner {
-    bitsize: u8,
-    next_expected_assignment: u128,
-}
-
-impl EnumVariantValueAssigner {
-    pub fn new(bitsize: u8) -> EnumVariantValueAssigner {
-        EnumVariantValueAssigner { bitsize, next_expected_assignment: 0 }
-    }
-    
-    fn max_value(&self) -> u128 {
-        (1u128 << self.bitsize) - 1
-    }
-
-    fn value_from_discriminant(&self, variant: &Variant) -> Option<u128> {
-        let discriminant = variant.discriminant.as_ref()?;
-        let discriminant_expr = &discriminant.1;
-        let variant_name = &variant.ident;
-
-        let Expr::Lit(ExprLit { lit: Lit::Int(int), .. }) = discriminant_expr else {
-            abort!(
-                discriminant_expr, 
-                "variant `{}` is not a number", variant_name; 
-                help = "only literal integers currently supported"
-            )
-        };
+/// attempts to extract the bitsize from a type token named `uN` or `bool`.
+/// should return `Result` instead of `Option`, if we decide to add more descriptive error handling.
+/// XXX: this strategy seems fragile.
+pub fn bitsize_from_type_token(path: &Path) -> Option<BitSize> {
```

Comment:
We only use this for better compile errors, right?

Reply:
as far as I'm aware, yes. 

the default compile error is actually pretty decent (something like `mismatched types expected struct bilge::prelude::UInt<_, 4> found struct bilge::prelude::UInt<_, 5>` and `expected 5, found 4` (I assume the second one is because rustc notices the expected and actual only differ by one character?). however the token it highlights is the `FromBits` and not the field inside the fallback variant.